internal/template-bundle/bundle.go

```diff
@@ -51,12 +53,31 @@ func readTemplates(filename string) ([]templatev1.Template, error) {
 		if err != nil {
 			return nil, err
 		}
-		if template.Name != "" {
+		if template.Name == "" {
+			continue
+		}
+		templateArch, retrieveTemplateArchErr := retrieveTemplateArch(template)
+		if retrieveTemplateArchErr != nil {
+			return nil, err
+		}
+		if templateArch == runtime.GOARCH {
 			bundle = append(bundle, template)
 		}
 	}
 }
 
+func retrieveTemplateArch(templ templatev1.Template) (string, error) {
+	var data map[string]interface{}
+	err := json.Unmarshal(templ.Objects[0].Raw, &data)
```

Comment:
Inline this into if below

Reply:
Can you please try to unmarshal that into a `VirtualMachine` instead of an interface. Perhaps even using [UnmarshalStrict](https://pkg.go.dev/sigs.k8s.io/yaml#UnmarshalStrict) if possible.

Reply:
While I wasn't able to implement it exactly as you envisioned, I believe I understand your idea.

Reply:
A template, in general, is not a valid `VirtualMachine` struct, so it cannot be unmarshalled. Because any field in the template can contain a variable that is substituted when the template is instantiated, something like: `${NAME}`. If this variable is on field that is not of type `string` the unmarshall would fail.

Reply:
@akrejcir @0xFelix I believe the trigger was the var data map[string]interface{}. Is that correct? If so, I have found an alternative solution.

Reply:
> If this variable is on field that is not of type string the unmarshall would fail.

I see, but `interface{}` stings my eye. Anything we can do here to get around it?

Reply:
[Solved](https://github.com/kubevirt/ssp-operator/pull/1150/files#diff-32b371bd23ccfa0083d4fcdfed2d7d28f755ca79dab2d6cca92e5b4fcbb16a0aR23)